x/clock/keeper/querier.go

```diff
@@ -24,13 +24,48 @@ func NewQuerier(k Keeper) Querier {
 func (q Querier) ClockContracts(stdCtx context.Context, _ *types.QueryClockContracts) (*types.QueryClockContractsResponse, error) {
 	ctx := sdk.UnwrapSDKContext(stdCtx)
 
-	p := q.keeper.GetParams(ctx)
+	contracts := q.keeper.GetAllContracts(ctx, false)
 
 	return &types.QueryClockContractsResponse{
-		ContractAddresses: p.ContractAddresses,
+		ContractAddresses: contracts,
+	}, nil
+}
+
+// JailedClockContracts returns contract addresses which have been jailed for erroring
+func (q Querier) JailedClockContracts(stdCtx context.Context, _ *types.QueryJailedClockContracts) (*types.QueryJailedClockContractsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(stdCtx)
+
+	contracts := q.keeper.GetAllContracts(ctx, true)
+
+	return &types.QueryJailedClockContractsResponse{
+		JailedContractAddresses: contracts,
 	}, nil
 }
 
+// ClockContract returns the clock contract information
+func (q Querier) ClockContract(stdCtx context.Context, req *types.QueryClockContract) (*types.QueryClockContractResponse, error) {
+	ctx := sdk.UnwrapSDKContext(stdCtx)
+
+	// Ensure the contract address is valid
+	if _, err := sdk.AccAddressFromBech32(req.ContractAddress); err != nil {
+		return nil, types.ErrInvalidAddress
+	}
+
+	// Check if the contract is jailed or unjailed
+	isUnjailed := q.keeper.IsClockContract(ctx, req.ContractAddress, false)
+	isJailed := q.keeper.IsClockContract(ctx, req.ContractAddress, true)
```

Comment:
this is confusing why it would check isUnjailed and isJailed? only isJailed needs checking?

Reply:
Resolved with new singular store